File: Cuarto/Simulacion/Proyecto/dist.r
# Wrapper function to invoke helloC with two arguments
dyn.load("prueba.so")
dist <- function(pobl) {
	mx <- matrix(0, nrow(pobl), nrow(pobl))
	result <- .C("dist", nrow=nrow(pobl), ncol=nrow(pobl), matrix=as.double(mx), pobl=as.double(pobl))
	return(result)
}

x<-matrix(seq(1,30), 10, 3)

y<-dist(x)

d<- matrix(y$matrix, 10, 10)

File: Cuarto/Simulacion/Proyecto/Proyecto_MarioGarcia_AlbertoCabello/dist.c
#include <math.h>
#include <stdio.h>
double duCl(double* pobl, int x, int y, int nrow){
	return sqrt( pow(pobl[x] - pobl[y], 2) + pow( pobl[x + nrow] - pobl[y+ nrow], 2));

}

void dist(int* nrow, int* ncol, double* matrix, double* pobl) {
	int i=0;
	int j=0;
	for(i=0; i< *nrow;i++){
		for(j=i; j< *nrow; j++){
			matrix[j+ (*ncol)*i] = matrix[i +(*ncol)*j]= duCl(pobl, i, j, *nrow);	
		}
	}
}



